### ISSUE TITLE
[Akamai] HTTP 401 when pagination due to bad signature calculation

### ISSUE THREAD

**Issue by andrewkroh** [elastic/integrations#9038]
**Background**

The Akamai SIEM API uses request signing for authorization.[^1][^2] The signing calculation includes the query parameters among other parts of the request.

**Problem**

When the integration makes a pagination request it gets a 401 response with a JSON body

```
{
  "title": "Not authorized",
  "status": 401,
  "detail": "The signature does not match",
  ...
}
```

**Cause**

The signature is calculated before the pagination transforms are executed. So when the `offset` query parameter is changed to paginate then the signature is invalidated.

The httpjson [docs](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-httpjson.html#response-pagination) explain the order: 

> All the transforms from request.transform will be executed and then response.pagination will be added to modify the next request as needed. 

[^1]: https://techdocs.akamai.com/developer/docs/authenticate-with-edgegrid#authentication-protocol-specification
[^2]:https://github.com/akamai/AkamaiOPEN-edgegrid-golang/blob/d417bd104d59eb9bf668da20c35f9bf899b65f90/pkg/edgegrid/signer.go#L44-L54

**Comment by andrewkroh** [comment]
I tested this problem under 8.10.4 and I could reproduce it. I also observed that sometimes it was not making all of the pagination requests. It would not make the next request until the interval (1h default) had passed.

Under 8.12.0 I didn't observe the 401. That implies the `last_offset` cursor is being saved after each pagination request such that it is accessible during this line, which would make the signature calculation correct.

https://github.com/elastic/integrations/blob/e974a509a0641758a6c587eea4fecf9795dfe8ed/packages/akamai/data_stream/siem/agent/stream/httpjson.yml.hbs#L30
